### PR TITLE
Add Sinhala E2E SLU Recipe 

### PR DESCRIPTION
**Pull request by karthik19967829**

### CODE REVIEW

**Comment by sw005320** [comment]
@siddhu001, can you review this PR?

File: egs2/sinhala/asr1/run.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+train_set="train"
+valid_set="valid"
+test_sets="test valid"
+
+asr_config=conf/train_asr.yaml
+
+./asr.sh \
+    --lang en \
+    --ngpu 1 \
+    --stage 11 \
```
Comment (ftshijt) [inline egs2/sinhala/asr1/run.sh:17]: could you modify the stage number to be the very start

File: egs2/sinhala/asr1/local/score.sh [hunk 2]
```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
+# Copyright 2021  Karthik Ganesan
```
Comment (ftshijt) [inline egs2/sinhala/asr1/local/score.sh:2]: @siddhu001 seems to have a unified scoring function, can you try that?

File: egs2/sinhala/asr1/local/data_prep.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        utt2spk_f.truncate()
+        transcript_df = pd.read_csv(os.path.join(sinhala_root, "data", dir_dict[x]))
+        # lines = sorted(transcript_df.values, key=lambda s: s[0])
+        for row in transcript_df.values:
+            print(row[4], " ".join([ch for ch in row[3]]))
+            words = row[4].replace(" ", "_") + " " + " ".join([ch for ch in row[3]])
+            print(words)
+            path_arr = row[1].split("/")
+            utt_id = path_arr[-2] + "_" + path_arr[-1]
+            # print(utt_id + " " + words + "\n")
```
Comment (ftshijt) [inline egs2/sinhala/asr1/local/data_prep.py:85]: please remove the debugging lines

**Review by ftshijt — CHANGES_REQUESTED** [review]
Can you also update the egs2/README.md with the dataset information?

File: egs2/sinhala/asr1/conf/tuning/train_asr_transformer.yaml [hunk 4]
```diff
@@ -0,0 +1,25 @@
+# network architecture
```
[the remark this answers is not in the document]
Comment (karthik19967829) [reply egs2/sinhala/asr1/conf/tuning/train_asr_transformer.yaml]: Updated this , sorry for the delay

File: egs2/sinhala/asr1/run.sh (hunk 1 above)
Comment (karthik19967829) [reply egs2/sinhala/asr1/run.sh]: Updated

File: egs2/sinhala/asr1/local/score.sh (hunk 2 above)
Comment (karthik19967829) [reply egs2/sinhala/asr1/local/score.sh:2]: Sure , added symlink to the unified scoring file

File: egs2/sinhala/asr1/local/data_prep.py (hunk 3 above)
Comment (karthik19967829) [reply egs2/sinhala/asr1/local/data_prep.py]: Sure , removed

File: egs2/sinhala/asr1/local/score.sh (hunk 2 above)
Comment (ftshijt) [reply egs2/sinhala/asr1/local/score.sh:2]: Can you also remove the file?

Comment (karthik19967829) [reply egs2/sinhala/asr1/local/score.sh]: Yes sure updated

**Comment by karthik19967829** [comment]
> Can you merge the latest commit from the master branch to active the fixed CI test?

Done

**Comment by karthik19967829** [comment]
@ftshijt it would be great if you can guide on how to fix the above CI errors

**Comment by ftshijt** [comment]
Please look at the error message.
- `Error: egs2/TEMPLATE/asr1/conf/slurm.conf and egs2/sinhala/asr1/conf/slurm.conf differ: To solve: for f in egs2/*/*/conf/slurm.conf; do cp egs2/TEMPLATE/asr1/conf/slurm.conf ${f}; done`

**Comment by karthik19967829** [comment]
Thanks , updated the slurm.conf

**Comment by karthik19967829** [comment]
@ftshijt to does the command  ./ci/test_integration_espnet2.sh  re-create the above CI tests locally ?

**Comment by karthik19967829** [comment]
@ftshijt I think there is a CI error in the master `FAILED test/espnet2/asr/transducer/test_beam_search_transducer.py::test_transducer_beam_search[search_params1-gru]`

**Comment by ftshijt** [comment]
Many thanks for your contribution!

**Pull request merged**